File: SPRING/Modern_Hospital_Management_System/src/main/java/emranhss/com/Modern_Hospital_Management_System/service/DietPlanService.java
package emranhss.com.Modern_Hospital_Management_System.service;

import emranhss.com.Modern_Hospital_Management_System.entity.DietPlan;
import java.util.List;

public interface DietPlanService {
    DietPlan create(DietPlan dietPlan);
    DietPlan getById(Long id);
    List<DietPlan> getAll();
    List<DietPlan> getActive();
    List<DietPlan> getByDietType(String dietType);
    DietPlan update(Long id, DietPlan dietPlan);
    void delete(Long id);
    List<DietPlan> search(String keyword);
    long getActiveCount();
}

File: SPRING/Hospital_Management_system_angular/src/app/services/dietary/diet-plan.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { DietPlan } from '../../models/dietary/diet-plan.model';

@Injectable({ providedIn: 'root' })
export class DietPlanService {
  private api = environment.apiUrl + 'diet-plans';

  constructor(private http: HttpClient) {}

  getAll(): Observable<DietPlan[]> {
    return this.http.get<DietPlan[]>(this.api);
  }
  getById(id: number): Observable<DietPlan> {
    return this.http.get<DietPlan>(`${this.api}/${id}`);
  }
  getActive(): Observable<DietPlan[]> {
    return this.http.get<DietPlan[]>(`${this.api}/active`);
  }
  getByDietType(dietType: string): Observable<DietPlan[]> {
    return this.http.get<DietPlan[]>(`${this.api}/type/${dietType}`);
  }
  search(keyword: string): Observable<DietPlan[]> {
    return this.http.get<DietPlan[]>(`${this.api}/search`, { params: { keyword } });
  }
  getActiveCount(): Observable<number> {
    return this.http.get<number>(`${this.api}/count/active`);
  }
  create(data: DietPlan): Observable<DietPlan> {
    return this.http.post<DietPlan>(this.api, data);
  }
  update(id: number, data: DietPlan): Observable<DietPlan> {
    return this.http.put<DietPlan>(`${this.api}/${id}`, data);
  }
  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.api}/${id}`);
  }
}
